spec/integration/parser/compiler_spec.rb

```diff
@@ -153,6 +153,17 @@ class c inherits b { notify { hi: } }
     end
   end
 
+  context 'when working with $settings name space' do
+    include PuppetSpec::Compiler
+    it 'makes $settings::strict available as string' do
+      node = Puppet::Node.new("testing")
+      catalog = compile_to_catalog(<<-MANIFEST, node)
+          notify { 'test': message => $settings::strict == 'warning' }
```

Comment:
It tests the symbol case since the strict setting is a symbol `:off`, `:warning`, `:error`.
IIRC, there is another PR dealing with additional types - we missed handling booleans correctly. That PR adds more tests.